enhancements/network/on-prem-mutable-vips.md

```diff
@@ -0,0 +1,213 @@
+This work however only covered installation-time, leaving clusters originally installed as single-stack and later converted to dual-stack out of scope.
+
+This design is proposing a change which will allow adding a second pair of virtual IPs to clusters that became dual-stack only during their lifetime.
+
+## Motivation
+
+We have customers who installed clusters before dual-stack VIP feature existed. Currently they have no way of migrating because the new feature is avaialble only during initial installation of the cluster.
+
+### User Stories
+
+* As a deployer of a dual-stack OpenShift cluster, I want to access the API using both IPv4 and IPv6.
+
+* As a deployer of a dual-stack OpenShift cluster, I want to access the Ingress using both IPv4 and IPv6.
+
+### Goals
+
+* Allow adding a second pair of virtual IPs to an already installed dual-stack cluster.
+
+* Allow deleting a second pair of virtual IPs on an already installed dual-stack cluster.
+
+### Non-Goals
+
+* Modifying existing virtual IP configuration.
+
+* Modifying virtual IP configuration after second pair of VIPs has been added. We only want to cover "create" and "delete" operation and only for the second pair of VIPs.
+
+* Configuration of any VIPs beyond a second pair for the second IP stack. MetalLB is a better solution for creating arbitrary loadbalancers.
+
+## Proposal
+
+### Workflow Description
+
+The proposed worklflow after implementing the feature would look as described below
+
+1. Administrator of a dual-stack cluster with single-stack VIPs wants to add a second pair for the second IP stack configured.
+
+1. Administrator edits the Infrastructure CR named `cluster` by changing the `spec.platformSpec.[*].apiServerInternalIPs` and `spec.platformSpec.[*].ingressIPs` fields.
+
+1. Cluster Network Operator picks the modification of the object and compares values with `spec.platformStatus.[*].apiServerInternalIPs` and `spec.platformStatus.[*].ingressIPs`.
+
+1. After validating that the requested change is valid (i.e. conforms with the goals and non-goals), the change is propagated down to the keepalived template and configuration file.
+
+1. After keepalived configuration is changed, the service is restarted or reloaded to apply the changes.
+
+### API Extensions
+
+New fields will have to be added to the platform spec section for [baremetal](https://github.com/openshift/api/blob/938af62eda38e539488d6193e7af292e47a09a5e/config/v1/types_infrastructure.go#L694), [vsphere](https://github.com/openshift/api/blob/938af62eda38e539488d6193e7af292e47a09a5e/config/v1/types_infrastructure.go#L1089)
+and [OpenStack](https://github.com/openshift/api/blob/938af62eda38e539488d6193e7af292e47a09a5e/config/v1/types_infrastructure.go#L772). In the first implementation only baremetal will be covered.
```

Comment:
I'm good with starting on baremetal only, but we probably need a followup story/epic to track adding the other platforms that support dual stack.